### PR TITLE
CDAP-2322: Set formatSpec in prepareJob

### PR DESCRIPTION
**Pull request by rohitsinha54**
Tested it with source as Stream and sink as Table and KVTable.
Build: http://builds.cask.co/browse/CDAP-DT62-1

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/StreamBatchSource.java [hunk 1]
```diff
@@ -98,6 +98,18 @@ public StreamBatchSource(StreamBatchConfig streamBatchConfig) {
 
   @Override
   public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineConfigurer) {
+    pipelineConfigurer.addStream(new Stream(streamBatchConfig.name));
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/StreamBatchSource.java]: still want to check the schema here. If it's an invalid schema we want to fail early at adapter creation

Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/StreamBatchSource.java]: similar comment about duration and delay. If we can't parse, should fail here.

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/StreamBatchSource.java [hunk 2]
```diff
@@ -119,17 +131,6 @@ public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineC
     } else {
       this.formatSpec = null;
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/StreamBatchSource.java]: there is no need for formatSpec to be a class variable, should be local to this method.

**Comment by albertshau** [comment]
mostly comments about failing early

**Comment by rohitsinha54** [comment]
LGTM

**Pull request merged**